### ISSUE TITLE
Alert Notifications secure_settings passing bool when api expects string

### ISSUE THREAD

**Issue by saltxwater** [grafana/terraform-provider-grafana#234]
### Terraform Version
* Terraform: 0.14.6
* Terraform Grafana Provider: 1.12.0
* Grafana: 7.5.7

### Affected Resource(s)
* grafana_alert_notification

### Terraform Configuration Files
```hcl
terraform {
  backend "local" {
    path = "localterraform.tfstate"
  }
  required_providers {
    grafana = {
      source = "grafana/grafana"
      version = "~> 1.12.0"
    }
  }
}

provider "grafana" {
  url = "<HOST>"
  auth = "<API KEY>"
}

resource "grafana_alert_notification" "slack" {
  name                    = "Slack"
  type                    = "slack"
  uid                     = "myUid"

  settings = {
    "url"         = "My Url"
  }

  secure_settings = {
    "url" = true
  }

}

```

### Debug Output
grafana_alert_notification.slack: Creating...

Error: status: 400, body: [{"classification":"DeserializationError","message":"json: cannot unmarshal bool into Go struct field CreateAlertNotificationCommand.secureSettings of type string"}]

  on notification-channels.tf line 2, in resource "grafana_alert_notification" "slack":
   2: resource "grafana_alert_notification" "slack" {

https://gist.github.com/saltxwater/ee3577a0e7449f021a8361d6cc20780d

### Expected Behavior
The Alert Notification to be created

### Actual Behavior
The request to grafana failed and no alert notification was created

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
As demonstrated here this fails when creating a new resource if you specify secure_settings. Linked to this:
- If you don't specify secure_settings and the notification has been saved in the grafana ui then secureFields will have { "url": true }. Which causes terraform to see the resource as always modified even after subsequent applies. You then can't add the secure_setting as you get the same error.
- _The grafana api for alert notifications is inconsistent; when reading secureFields it is a bool but expects strings for create and update: https://github.com/grafana/grafana/blob/main/pkg/models/alert_notifications.go **Edit: This is not correct - secureFields is correct to be bool but it is doing the create on secureSettings which IS meant to be strings**_

**Comment by saltxwater** [comment]
Digging into this some more:
I'm a bit confused about what the format of secure_settings should be.

If you create the resource WITHOUT specifying secure_settings then the resource is created fine in grafana.
If you then go into the grafana ui, view the notification, and hit save then the PUT request is:
`{"id":33,"name":"Slack","type":"slack","sendReminder":false,"disableResolveMessage":false,"frequency":"15m","settings":{"uploadImage":false,"autoResolve":true,"httpMethod":"POST","severity":"critical","url":"","token":""},"secureSettings":{"url":"My Url"},"secureFields":{},"isDefault":false,"uid":"myUid","created":"2021-06-29T12:35:43+01:00","updated":"2021-06-29T12:35:43+01:00"}`
(Not setting "secureFields", but it IS setting secureSettings)

And the response:
`{"id":33,"uid":"myUid","name":"Slack","type":"slack","isDefault":false,"sendReminder":false,"disableResolveMessage":false,"frequency":"","created":"2021-06-29T12:35:43+01:00","updated":"2021-06-29T12:35:54+01:00","settings":{"autoResolve":true,"httpMethod":"POST","severity":"critical","token":"","uploadImage":false,"url":""},"secureFields":{"url":true}}`

But then if you try again (in the ui) to save it then the request is:
`{"id":33,"name":"Slack","type":"slack","sendReminder":false,"disableResolveMessage":false,"frequency":"15m","settings":{"uploadImage":false,"autoResolve":true,"httpMethod":"POST","severity":"critical","token":"","url":""},"secureSettings":{},"secureFields":{"url":true},"isDefault":false,"uid":"myUid","created":"2021-06-29T12:35:43+01:00","updated":"2021-06-29T12:35:54+01:00"}`

(Not setting "secureSettings", but it IS setting secureFields).

It's a pain of the grafana api, but I don't think this terraform provider is handling it correctly.

**Comment by saltxwater** [comment]
https://github.com/grafana/terraform-provider-grafana/blob/master/grafana/resource_alert_notification.go#L256

When the provider creates the alert notification it is using secureSettings (but passing bools when I think this should actually be strings / the value).

When reading it is using secureFields (which is the only thing available to it)
https://github.com/grafana/terraform-provider-grafana/blob/master/grafana/resource_alert_notification.go#L162

Also, since the response now has the url as empty, I think the provider sees the url has having changed so terraform continues to say changes are needed.

Hope this helps!

**Comment by aholyoake-bc** [comment]
I'm seeing this when applying the same slack alert notifications setup over and over.
An excerpt from `terraform show -json tfplan`



```
"resource_drift": [
...
      "change": {
        "actions": [
          "update"
        ],
        "before": {
          "disable_resolve_message": false,
          "frequency": "",
          "id": "1",
          "is_default": false,
          "name": "#monitoring-dev-test-team",
          "secure_settings": {
            "recipient": "#monitoring-dev-test-team",
            "token": "$TOKEN",
            "username": "Grafana Alerting"
          },
          "send_reminder": false,
          "settings": {},
          "type": "slack",
          "uid": "XXX"
        },
        "after": {
          "disable_resolve_message": false,
          "frequency": "",
          "id": "1",
          "is_default": false,
          "name": "#monitoring-dev-test-team",
          "secure_settings": {
            "recipient": "true",
            "token": "true",
            "username": "true"
          },
          "send_reminder": false,
          "settings": {},
          "type": "slack",
          "uid": "XXX"
        },
        "before_sensitive": {
          "secure_settings": {
            "token": true
          },
          "settings": {}
        },
        "after_sensitive": {
          "secure_settings": {
            "token": true
          },
          "settings": {}
        }
      }


]


"resource_changes": [
...
      "change": {
        "actions": [
          "update"
        ],
        "before": {
          "disable_resolve_message": false,
          "frequency": "",
          "id": "1",
          "is_default": false,
          "name": "#monitoring-dev-test-team",
          "secure_settings": {
            "recipient": "true",
            "token": "true",
            "username": "true"
          },
          "send_reminder": false,
          "settings": {},
          "type": "slack",
          "uid": "XXX"
        },
        "after": {
          "disable_resolve_message": false,
          "frequency": "",
          "id": "1",
          "is_default": false,
          "name": "#monitoring-dev-test-team",
          "secure_settings": {
            "recipient": "#monitoring-dev-test-team",
            "token": "$TOKEN",
            "username": "Grafana Alerting"
          },
          "send_reminder": false,
          "settings": {},
          "type": "slack",
          "uid": "XXX"
        },
        "after_unknown": {},
        "before_sensitive": {
          "secure_settings": true,
          "settings": {}
        },
        "after_sensitive": {
          "secure_settings": true,
          "settings": {}
        }
      }
]
```

i.e. the provider is reading the secure_settings as 
```
 "secure_settings": {
            "recipient": "true",
            "token": "true",
            "username": "true"
          }
```
and trying to write them (correctly) as 
```
          "secure_settings": {
            "recipient": "#monitoring-dev-test-team",
            "token": "$TOKEN",
            "username": "Grafana Alerting"
          },
```

**Comment by inkel** [comment]
The [documentation for `grafana_alert_notification`](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/alert_notification#optional) has the following for both `settings` and `secure_settings`:

> * `secure_settings` (Map of String, Sensitive) Additional secure settings, for full reference lookup Grafana Supported Settings documentation.
> * `settings` (Map of String) Additional settings, for full reference see Grafana HTTP API documentation.

So the `.tf` files should indeed use strings instead of booleans.

I can confirm the issue is indeed an issue with the provider. I'll be working on a fix.

**Comment by inkel** [comment]
This has been fixed in #322. Once that one gets merged and a new release is done it should be ready to mark as closed.

**Comment by julienduchesne** [comment]
This has been released in `v1.18.0`
